src/Extension/CoreExtension.php

```diff
@@ -1300,6 +1302,65 @@ function twig_test_iterable($value)
     return $value instanceof \Traversable || \is_array($value);
 }
 
+/**
+ * Checks if a variable is a sequence.
+ *
+ *    {# evaluates to true if the foo variable is a list or a traversable object #}
+ *    {% if foo is sequence %}
+ *        {# ... #}
+ *    {% endif %}
+ *
+ * @param mixed $value A variable
+ *
+ * @return bool true if the value is a sequence
+ */
+function twig_test_sequence($value)
+{
+    return $value instanceof \Traversable || (\is_array($value) && \array_is_list($value));
+}
+
+
+/**
+ * Checks if a variable is a mapping.
+ *
+ *    {# evaluates to true if the foo variable is an array but not a list #}
+ *    {% if foo is mapping %}
+ *        {# ... #}
+ *    {% endif %}
+ *
+ * @param mixed $value A variable
+ *
+ * @return bool true if the value is a mapping
+ */
+function twig_test_mapping($value)
+{
+    return \is_array($value) && ((0 === \count($value)) || !\array_is_list($value));
+}
+
+/**
+ * Checks whether a given array is a list.
+ *
+ * Polyfill for PHP 7.4 to 8.0.
+ * From https://www.php.net/manual/en/function.array-is-list.php#127044
+ *
+ * @param mixed $array The array being evaluated.
+ *
+ * @return bool true if array is a list, false otherwise
+ */
+if (!function_exists('array_is_list')) {
+    function array_is_list(array $array)
```

Comment:
`symfony/polyfill-php81` should be used instead of creating the polyfill here.
But that will not be necessary if you change the implementation to support any `iterable` (previous comment).

Reply:
this should indeed depend on `symfony/polyfill-php81` instead of providing its own untested polyfill.